### PR TITLE
Adding Cooler Master PSU support

### PR DESCRIPTION
**Pull request by caerffili**
Adding Cooler Master PSU support. The following PSUs are supported

- V550 GOLD i MULTI
- V650 GOLD i MULTI
- V750 GOLD i MULTI
- V850 GOLD i MULTI
- V550 GOLD i 12VO
- V650 GOLD i 12VO
- V750 GOLD i 12VO
- V850 GOLD i 12VO
- V650 PLATINUM i 12VO
- V750 PLATINUM i 12VO
- V850 PLATINUM i 12VO
- FANLESS 1300

Tested on V850 Gold i.

image

This is based on work by Jannis234

https://github.com/Jannis234/cm-psu

### CODE REVIEW

File: LibreHardwareMonitorLib/LibreHardwareMonitorLib.csproj [hunk 1]
```diff
@@ -48,4 +48,7 @@
       <Pack>True</Pack>
     </None>
   </ItemGroup>
+  <ItemGroup>
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/LibreHardwareMonitorLib.csproj:51]: Not needed.

File: LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsuGroup.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsuGroup.cs:1]: Format doc.

File: LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsu.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+using System.Xml.Linq;
+using HidSharp;
+using HidSharp.Utility;
+using LibreHardwareMonitor.Hardware.Motherboard.Lpc;
+using LibreHardwareMonitor.Hardware.Psu.Corsair;
+using static LibreHardwareMonitor.Interop.Ftd2xx;
+
+namespace LibreHardwareMonitor.Hardware.Psu.CoolerMaster;
+
+
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsu.cs:32]: Format doc / follow code style.

**Review by PhyxionNL — CHANGES_REQUESTED** [review]

File: LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsu.cs (hunk 3 above)
Comment (caerffili) [reply LibreHardwareMonitorLib/Hardware/Psu/CoolerMaster/CoolerMasterPsu.cs:32]: Is the code style documented anywhere? I couldn't find it.

If not, I'll study some other code and try to work it out.

Thanks :)

**Comment by PhyxionNL** [comment]
Closing this due to stale and also due to Update not doing any updates like it should.

**Pull request closed without merging**